ospf6d/ospf6_lsa.c

```diff
@@ -551,27 +551,49 @@ void ospf6_lsa_show_dump(struct vty *vty, struct ospf6_lsa *lsa,
 {
 	uint8_t *start, *end, *current;
 	char byte[4];
+	char *header_str;
+	char adv_router[64];
+	char id[64];
+	json_object *json;
 
 	start = (uint8_t *)lsa->header;
 	end = (uint8_t *)lsa->header + ntohs(lsa->header->length);
 
-	if (use_json)
-		return;
+	if (use_json) {
+		json = json_object_new_object();
+		header_str = (char *)(malloc((2 * (end - start)) + 1));
 
-	vty_out(vty, "\n");
-	vty_out(vty, "%s:\n", lsa->name);
+		inet_ntop(AF_INET, &lsa->header->id, id, sizeof(id));
+		inet_ntop(AF_INET, &lsa->header->adv_router, adv_router,
+			  sizeof(adv_router));
 
-	for (current = start; current < end; current++) {
-		if ((current - start) % 16 == 0)
-			vty_out(vty, "\n        ");
-		else if ((current - start) % 4 == 0)
-			vty_out(vty, " ");
+		for (current = start; current < end; current++)
+			snprintf(&header_str[2 * (current - start)], 3, "%02x",
+				 *current);
+		header_str[2 * (end - start)] = '\0';
 
-		snprintf(byte, sizeof(byte), "%02x", *current);
-		vty_out(vty, "%s", byte);
-	}
+		json_object_string_add(json, "linkStateId", id);
+		json_object_string_add(json, "advertisingRouter", adv_router);
+		json_object_string_add(json, "header", header_str);
+		json_object_array_add(json_array, json);
+
+		free(header_str);
+	} else {
+		vty_out(vty, "\n");
+		vty_out(vty, "%s:\n", lsa->name);
```

Comment:
can you print the leading `\n` as part of this call?

Reply:
Fixed.